HELICS-src/src/helics/core/CommonCore.cpp:2878:22: error: call to unavailable function       'any_cast'

Error when building with clang++ on v1.0.1

```bash
$ make
Scanning dependencies of target helics_core
[  0%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/CommonCore.cpp.o
/Users/$USER/Documents/GitRepos/HELICS-src/src/helics/core/CommonCore.cpp:2878:22: error: call to unavailable function
      'any_cast':
            auto M = stx::any_cast<std::shared_ptr<FilterOperator>> (std::move(*op));
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/any:542:12: note: candidate
      function [with _ValueType = std::__1::shared_ptr<helics::FilterOperator>] has been explicitly made unavailable
_ValueType any_cast(any && __v)
           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/any:511:12: note: candidate
      function [with _ValueType = std::__1::shared_ptr<helics::FilterOperator>] has been explicitly made unavailable
_ValueType any_cast(any const & __v)
           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/any:527:12: note: candidate
      function not viable: expects an l-value for 1st argument
_ValueType any_cast(any & __v)
           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/any:558:1: note: candidate
      function not viable: no known conversion from 'typename remove_reference<any &>::type' (aka 'std::experimental::fundamentals_v1::any')
      to 'const std::experimental::fundamentals_v1::any *' for 1st argument
any_cast(any const * __any) _NOEXCEPT
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/experimental/any:567:1: note: candidate
      function not viable: no known conversion from 'typename remove_reference<any &>::type' (aka 'std::experimental::fundamentals_v1::any')
      to 'std::experimental::fundamentals_v1::any *' for 1st argument
any_cast(any * __any) _NOEXCEPT
^
1 error generated.
make[2]: *** [src/helics/core/CMakeFiles/helics_core.dir/CommonCore.cpp.o] Error 1
make[1]: *** [src/helics/core/CMakeFiles/helics_core.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/helics/common/CMakeLists.txt
+++ b/src/helics/common/CMakeLists.txt
@@ -89,5 +89,9 @@ if (HELICS_HAVE_ZEROMQ)
    target_include_directories (helics_common PRIVATE "${ZeroMQ_INCLUDE_DIR}")
 endif(HELICS_HAVE_ZEROMQ)
 
+if (APPLE)
+	target_compile_definitions(helics_common PUBLIC "-DSTX_NO_STD_ANY=1")
+endif(APPLE)
+
 install(FILES ${helics_public_common} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/helics/common COMPONENT headers)
 

--- a/src/helics/core/CMakeLists.txt
+++ b/src/helics/core/CMakeLists.txt
@@ -174,6 +174,10 @@ IF (VERSION_OPTION)
 	ENDIF (MSVC)
 ENDIF(VERSION_OPTION)
 
+if (APPLE)
+	target_compile_definitions(helics_core PUBLIC "-DSTX_NO_STD_ANY=1")
+endif(APPLE)
+
 if (${MPI_C_FOUND})
 source_group("mpi" FILES ${MPI_SOURCE_FILES} ${MPI_HEADER_FILES})
 endif (${MPI_C_FOUND})

